### PR TITLE
Feature/cleanup base file

### PR DESCRIPTION
**Pull request by Alexanderlacuna**
This PR cleans up the base file. It includes the following changes:
* [] Move the global search to the index page
* [] Remove custom JS from the base file and add it to the index file
* [] Move the global search to the index page
* [] fix incorrect html tags
* []  formatting for base file

### CODE REVIEW

**Review by BonfaceKilz — APPROVED** [review]
@Alexanderlacuna I've tested out your work locally and looked at the individual commits.  Great work for the general clean-up.  We can have this merged when you are ready.  Thanks!

File: gn2/wqflask/templates/base.html [hunk 1]
```diff
@@ -33,262 +46,297 @@
...
+                                    </li>
+                                    <li>
+                                        <a href="https://github.com/genenetwork/sysmaintenance">System Maintenance Code</a>
+                                    </li>
+                                    <li>
+                                        <a href="https://github.com/genenetwork/genenetwork2/blob/testing/doc/API_readme.md">REST API Documentation</a>
+                                    </li>
+                                </ul>
                             </li>
+                            {% if g.user_session.logged_in %}
```
Comment (jnduli) [inline gn2/wqflask/templates/base.html]: nitpick: `if logged_in() `? The advantage is that the function also checks if the token is valid

**Review by jnduli — CHANGES_REQUESTED** [review]
My main concern regards losing the search button in may other pages. I've done a `git grep base.html` and there 125 files that will be affected. 

```
╰─$ git --no-pager grep "base.html" | wc -l
125
```

Here's an example scenario:
image
 
 Are we ok with this? I think if not, we could just change these pages to extend `index.html` instead.

**Comment by BonfaceKilz** [comment]
@jnduli 

> My main concern regards losing the search button in may other pages. I've done a `git grep base.html` and there 125 files that will be affected.
> 
> ```
> ╰─$ git --no-pager grep "base.html" | wc -l
> 125
> ```
> 
> Here's an example scenario:
> [...]
> 
> Are we ok with this? I think if not, we could just change these pages to extend `index.html` instead.

@jnduli thanks for catching the above.

**Comment by Alexanderlacuna** [comment]
@jnduli This may need clarification. CC @BonfaceKilz, CC @zsloan. Although I don’t think we need to include global search for all pages, especially those that are not related to it, such as gnqna, docs, etc., in my opinion, it should only be implemented for search-related functionalities and pages that render search results. In this PR, I have only added it for the index page and extended the functionality to the gsearch result pages.

File: gn2/wqflask/templates/base.html (hunk 1 above)
Comment (Alexanderlacuna) [reply gn2/wqflask/templates/base.html]: Done!

**Review by jnduli — APPROVED** [review]

**Review by BonfaceKilz — APPROVED** [review]

**Pull request merged**